src/uu/shuf/src/shuf.rs

```diff
@@ -10,132 +10,165 @@
+                    show_error!("invalid line count: '{}'", count.trim());
                     return 1;
                 }
-                match parse_range(range) {
-                    Ok(m) => Mode::InputRange(m),
-                    Err(msg) => {
-                        crash!(1, "{}", msg);
-                    }
-                }
-            }
-            None => {
-                if echo {
-                    Mode::Echo
-                } else {
-                    if matches.free.is_empty() {
-                        matches.free.push("-".to_owned());
-                    } else if matches.free.len() > 1 {
-                        show_error!("extra operand '{}'", &matches.free[1][..]);
-                    }
-                    Mode::Default
-                }
-            }
-        };
-        let repeat = matches.opt_present("repeat");
-        let sep = if matches.opt_present("zero-terminated") {
+            },
+            None => usize::MAX,
+        },
+        output: matches.value_of(OPT_OUTPUT).map(String::from),
+        random_source: matches.value_of(OPT_RANDOM_SOURCE).map(String::from),
+        repeat: matches.is_present(OPT_REPEAT),
+        sep: if matches.is_present(OPT_ZERO_TERMINATED) {
             0x00_u8
         } else {
             0x0a_u8
-        };
-        let count = match matches.opt_str("head-count") {
-            Some(cnt) => match cnt.parse::<usize>() {
-                Ok(val) => val,
-                Err(e) => {
-                    show_error!("'{}' is not a valid count: {}", cnt, e);
-                    return 1;
-                }
-            },
-            None => MAX_USIZE,
-        };
-        let output = matches.opt_str("output");
-        let random = matches.opt_str("random-source");
+        },
+    };
 
-        match mode {
-            Mode::Echo => {
-                // XXX: this doesn't correctly handle non-UTF-8 cmdline args
-                let mut evec = matches
-                    .free
-                    .iter()
-                    .map(String::as_bytes)
-                    .collect::<Vec<&[u8]>>();
-                find_seps(&mut evec, sep);
-                shuf_bytes(&mut evec, repeat, count, sep, output, random);
-            }
-            Mode::InputRange((b, e)) => {
-                let rvec = (b..e).map(|x| format!("{}", x)).collect::<Vec<String>>();
-                let mut rvec = rvec.iter().map(String::as_bytes).collect::<Vec<&[u8]>>();
-                shuf_bytes(&mut rvec, repeat, count, sep, output, random);
-            }
-            Mode::Default => {
-                let fdata = read_input_file(&matches.free[0][..]);
-                let mut fdata = vec![&fdata[..]];
-                find_seps(&mut fdata, sep);
-                shuf_bytes(&mut fdata, repeat, count, sep, output, random);
-            }
+    match mode {
+        Mode::Echo(args) => {
+            // XXX: this doesn't correctly handle non-UTF-8 cmdline args
```

Comment:
is this comment still relevant with clap?

Reply:
```
$ ./target/debug/coreutils shuf -e こんにちは -e t
t
こんにちは
```
No, it seems like it is no longer relevant. I'll remove it. Thanks!